### PR TITLE
add exception handeling for warning serialization issues

### PR DESCRIPTION
**Pull request by BrandonHoffman**
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs:

- [x] Make sure to include reasonable tests for your change if necessary

- [x] We use [towncrier](https://pypi.python.org/pypi/towncrier) for changelog management, so please add a *news* file into the `changelog` folder following these guidelines:
  * Name it `$issue_id.$type` for example `588.bugfix`;
  * If you don't have an issue_id change it to the PR id after creating it
  * Ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example:

    ```
    Fix issue with non-ascii contents in doctest text files.
    ```

### CODE REVIEW

File: xdist/workermanage.py [hunk 1]
```diff
@@ -426,7 +426,10 @@ def unserialize_warning_message(data):
     if data["message_module"]:
         mod = importlib.import_module(data["message_module"])
         cls = getattr(mod, data["message_class_name"])
-        message = cls(*data["message_args"])
+        try:
+            message = cls(*data["message_args"])
+        except TypeError:
+            message = Warning(data["message_str"])
```
Comment (RonnyPfannschmidt) [inline xdist/workermanage.py]: lets add the original warning  glass name here for a good measure

File: testing/acceptance_test.py [hunk 2]
```diff
@@ -413,6 +413,8 @@ def test_warnings(self, testdir, n, warn_type):
         elif warn_type == "pytest":
             warn_code = """request.config.warn('', 'this is a warning',
                            fslocation=py.path.local())"""
+        elif warn_type == "invalid":
```
Comment (nicoddemus) [inline testing/acceptance_test.py]: Thanks @BrandonHoffman, but this test passes for me on current `master`:

```
======================== test session starts ========================
platform win32 -- Python 3.6.3, pytest-3.8.1, py-1.5.2, pluggy-0.7.1 -- x:\pytest-xdist\.env36\scripts\python.exe
cachedir: .pytest_cache
rootdir: x:\pytest-xdist, inifile: tox.ini
plugins: forked-0.2, xdist-1.22.6.dev1+g96af694.d20180818
collected 67 items / 61 deselected

testing/acceptance_test.py::TestTerminalReporting::test_warnings[pytest--n0] PASSED [ 16%]
testing/acceptance_test.py::TestTerminalReporting::test_warnings[pytest--n1] PASSED [ 33%]
testing/acceptance_test.py::TestTerminalReporting::test_warnings[builtin--n0] PASSED [ 50%]
testing/acceptance_test.py::TestTerminalReporting::test_warnings[builtin--n1] PASSED [ 66%]
testing/acceptance_test.py::TestTerminalReporting::test_warnings[invalid--n0] PASSED [ 83%]
testing/acceptance_test.py::TestTerminalReporting::test_warnings[invalid--n1] PASSED [100%]

============== 6 passed, 61 deselected in 1.56 seconds ==============
```

So this is not really reproducing the problem... 😕

Comment (nicoddemus) [reply testing/acceptance_test.py]: I've written a test which reproduces the issue, will push shortly. 👍

**Comment by BrandonHoffman** [comment]
@nicoddemus thanks I was really strugling to get a test case that hit the issue did not since the testdir was not in the path.....

I did not realize there was a

```
testdir.syspathinsert()
```

**Pull request merged**